src/openforms/submissions/tasks.py

```diff
@@ -0,0 +1,232 @@
+import logging
+from datetime import timedelta
+
+from django.db.models import (
+    Case,
+    CharField,
+    DateTimeField,
+    DurationField,
+    ExpressionWrapper,
+    F,
+    IntegerField,
+    Value,
+    When,
+)
+from django.utils import timezone
+
+from ..celery import app
+from ..config.models import GlobalConfiguration
+from ..submissions.constants import RemovalMethods
+from .constants import RegistrationStatuses
+from .models import Submission
+
+logger = logging.getLogger(__name__)
+
+
+@app.task
+def delete_submissions():
+    logger.debug("Deleting submissions")
+
+    config = GlobalConfiguration.get_solo()
+
+    Submission.objects.annotate(
+        removal_method=Case(
+            When(
+                form__successful_submissions_removal_method=None,
+                then=Value(config.successful_submissions_removal_method),
+            ),
+            default=F("form__successful_submissions_removal_method"),
+            output_field=CharField(),
+        ),
+        removal_limit=Case(
+            When(
+                form__successful_submissions_removal_limit=None,
+                then=Value(config.successful_submissions_removal_limit),
+            ),
+            default=F("form__successful_submissions_removal_limit"),
+            output_field=IntegerField(),
+        ),
+        time_since_creation=ExpressionWrapper(
+            Value(timezone.now(), DateTimeField()) - F("created_on"),
+            output_field=DurationField(),
+        ),
+    ).filter(
```

Comment:
Can we abstract that away in a custom queryset method? Should be something like:

```py
Submission.objects.annotate_removal_fields("successful_submissions_removal_method", "successful_submissions_removal_limit")
```

The custom method then abstracts getting the default values and building the Case/When query, so you can also unit-test it independently.